### PR TITLE
Update simplicity

### PR DESCRIPTION
**Pull request by uncomputable**
Update rust-simplicity to the latest version. Write documentation for the new jets. Fix documentation issues. Include codegen in CI.

### CODE REVIEW

File: src/compile.rs [hunk 1]
```diff
@@ -342,7 +352,11 @@ impl Call {
 /// The fold `(fold f)_n : E^(<2^n) × A → A`
 /// takes the list of type `E^(<2^n)` and an initial accumulator of type `A`,
 /// and it produces the final accumulator of type `A`.
-fn list_fold(bound: NonZeroPow2Usize, f: &ProgNode) -> Result<ProgNode, simplicity::types::Error> {
+fn list_fold(
+    ctx: &simplicity::types::Context,
+    bound: NonZeroPow2Usize,
+    f: &ProgNode,
+) -> Result<ProgNode, simplicity::types::Error> {
```
Comment (apoelstra) [inline src/compile.rs:359]: In 7368e4597e17d9528c5bff43746c02135ce38cc6:

Here and elsewhere you should be able to take the ctx out of `f` so you don't need an extra parameter and you won't risk calling the method incorrectly.

**Comment by apoelstra** [comment]
3436b283fe5bc3a89b952ea618222113951e9aad looks good other than the ctx cleanups in the first commit.

File: src/compile.rs (hunk 1 above)
Comment (uncomputable) [reply src/compile.rs]: Great catch. Thanks for pointing this out.

**Comment by uncomputable** [comment]
Removed superfluous `ctx` function parameters.

**Comment by apoelstra** [comment]
In 98d23965fa6d997dc53512ba11e0278429ee751b:

Need to also update the `codegen` crate which doesn't build on this commit.

**Comment by uncomputable** [comment]
Merged the first two commits. Each commit should compile now.

**Review by apoelstra — APPROVED** [review]
ACK 0ccfa9b00395c1261f1a028d9c448b4c22c58c1d successfully ran local tests

**Pull request merged**